lib/ddtrace/context.rb

```diff
@@ -13,110 +13,84 @@ module Datadog
   # \Context, it will be related to the original trace.
   #
   # This data structure is thread-safe.
-  # rubocop:disable Metrics/ClassLength
   class Context
     # 100k spans is about a 100Mb footprint
     DEFAULT_MAX_LENGTH = 100_000
 
     attr_reader :max_length
 
-    # Initialize a new thread-safe \Context.
+    # Initialize a new \Context.
     def initialize(options = {})
-      @mutex = Mutex.new
       # max_length is the amount of spans above which, for a given trace,
       # the context will simply drop and ignore spans, avoiding high memory usage.
       @max_length = options.fetch(:max_length, DEFAULT_MAX_LENGTH)
       reset(options)
     end
 
     def trace_id
-      @mutex.synchronize do
-        @parent_trace_id
-      end
+      @parent_trace_id
     end
 
     def span_id
-      @mutex.synchronize do
-        @parent_span_id
-      end
+      @parent_span_id
     end
 
-    def sampling_priority
-      @mutex.synchronize do
-        @sampling_priority
-      end
-    end
+    attr_reader :sampling_priority
 
-    def sampling_priority=(priority)
-      @mutex.synchronize do
-        @sampling_priority = priority
-      end
-    end
+    attr_writer :sampling_priority
 
     # Return the last active span that corresponds to the last inserted
     # item in the trace list. This cannot be considered as the current active
     # span in asynchronous environments, because some spans can be closed
     # earlier while child spans still need to finish their traced execution.
-    def current_span
-      @mutex.synchronize do
-        return @current_span
-      end
-    end
+    attr_reader :current_span
 
     # Add a span to the context trace list, keeping it as the last active span.
     def add_span(span)
```

Comment:
AFAIK context is currently always thread-local. As such it doesn't need to be synchronized